sumologic/resource_sumologic_polling_source.go

```diff
@@ -75,19 +75,69 @@ func resourceSumologicPollingSource() *schema.Resource {
 		MaxItems: 1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
+				"type": {
+					Type:         schema.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringInSlice([]string{"S3BucketPathExpression", "CloudWatchPath"}, false),
+				},
 				"bucket_name": {
 					Type:     schema.TypeString,
-					Required: true,
+					Optional: true,
 					ForceNew: false,
```

Comment:
@seansain what about the `ForceNew:false` which is already being used at couple of places in this file prior to this change?

Reply:
if it works as expected, remove those too